aerogearcatalog/aerogear-digger-apb
# Allow configure volume size

This PR fixed a few issues while try to provision a mobile CI/CD service in an RHPDS instance.

This PR has been built and pushed to docker hub. To verify:

1. Make sure you have a RHPDS instance running
2. ssh into the bastion machine and make sure you are running as a system admin user for the openshift cluster.
3. Run
    ```
    oc edit configmap broker-config -n openshift-ansible-service-broker
    ```
    and add this section to `registry` list
    ```
    - type: dockerhub
        name: test
        url: https://registry.hub.docker.com
        org: weilee
        tag: latest
        white_list:
          - '.*-apb$'
    ```
    also make sure change the `image_pull_policy` in `openshift` section to `Always`
4. Restart asb by running `oc rollout latest asb -n openshift-ansible-service-broker` to make sure the latest images are fetched
5. Refresh catalog:
    ```
    oc get clusterservicebroker ansible-service-broker -o=json > broker.json
    oc delete clusterservicebroker ansible-service-broker
    oc create -f broker.json
    ```

Then go to the mobile service catalog page. You should see a new "Mobile CI/CD Test" service. Try provision that service and make sure update the configuration options during the provision step.

NOTE: there is a known issue that after the provision is finished, the android sdk may not be fully installed. To check if the android sdk install is completed:

*  there should be a `android-sdk` pod running after the provision is completed
* use the terminal and run `ls /opt/android-sdk-linux/platforms` and you should see `android-27` in the output. 

If you don't see it or get any error, you will need to reinstall the android sdk. To install manually:

* there should be a `android-sdk` pod running after the provision is completed
* use the terminal and run:
    ```
    cd /opt/tools/
    ./androidctl-sync ./sdk.cfg
    ```
    wait for it to finish and you should have the sdk installed properly

Now you can create a build and build it.

To try kryptowire integration, you can use this repo: https://github.com/wei-lee/android-showcase-template.git

ping @maleck13 @johnfriz @evanshortiss
